### DISCUSSION TITLE
Using JDBI to generate sql

### DISCUSSION THREAD

**Discussion by marceloverdijk** [jdbi/jdbi#2661] [Q&A] [answered]
Hi,

I'm using JDBI to populate a database from raw json files and it works perfectly. 
Also performance is awesome.

I'm using the Dao pattern and resolve the files with sql statements from the classpath like:

``` 
@UseClasspathSqlLocator
interface MyDao {

    @SqlScript("create_schema")
    fun createSchema()

    @SqlBatch("insert_continent")
    fun insertContinents(@BindBean continents: List<Continent>)

    ..
```

e.g.:

```
INSERT INTO continent
( id
, code
, name
)
VALUES
( :id
, :code
, :name
);
```

Now besides populating this database, I would like to create a *.sql with the plain SQL statements which were executed.
As I'm not using anything special this *.sql file could be used to import in various databases like SQLite, MySQL, PostgreSQL, ..

As I already have the JDBI implementation in place and the sql files (see example above) to insert the data, I wonder if i could use JDBI to generate this *.sql file by:
- getting the raw sql to be send to the database (in the correct order)
- avoid the statement to actually send to the database (but this is not that important; as I could use a database in memory and just discard it)

Would this be possible with JDBI or should I look for something else?

**Comment by stevenschlansker** [comment, marked as answer]
Hi @marceloverdijk , Jdbi currently sees generating SQL as out of scope for this project. There are already other libraries like jOOQ that do this and we don't see the need to duplicate their work at this time.

**Reply by marceloverdijk** [reply]
Thx for confirming.
